File: Trabajo_9/CPP/Actividades/Actividad_3.cpp
#include <iostream>
using namespace std;

class Figura{
public:
    virtual void calcularArea() = 0;
};

class Circulo : public Figura {
private:
    int radio;
    const float pi = 3.14;
public:

    Circulo(int r){
        this -> radio = r;
    }

    void calcularArea() override{
        cout << "Area: " << pi*(radio * radio) << endl;
    }
};

class Rectangulo : public Figura{
private:
    int base;
    int altura;

public:

    Rectangulo(int b, int h){
        this -> base = b;
        this -> altura = h;
    }

    void calcularArea() override{
        cout << "Area: " << base * altura << endl;
    }
};

class Triangulo : public Figura{
private:
    int base;
    int altura;
public:

    Triangulo(int b, int h){
        this -> base = b;
        this -> altura = h;
    }

    void calcularArea() override{
        cout << "Area: " << (base * altura)/2 << endl;
    }
};

int main(){
    //Trabajamos dinamicamente con punteros
    Figura* circulito = new Circulo(2);
    Figura* rectangulito = new Rectangulo(2,5);
    Figura* triangulito = new Triangulo(10,80);


    Figura *arreglo[3] = {circulito, rectangulito, triangulito};

    for (Figura *p : arreglo){
        p->calcularArea();
    }
    return 0;




}

File: Trabajo_9/PY/Actividades/Actividad_3.py
from abc import ABC, abstractmethod
class Figura(ABC):
    def calcularArea(self):
        pass

class Circulo(Figura):
    def __init__(self,radio):
        self.radio=radio
        self.pi=3.141592

    def calcularArea(self):
        print(f"Area de la circulo {self.pi*(self.radio * self.radio)}\n")

class Rectangulo(Figura):
    def __init__(self,base,altura):
        self.base=base
        self.altura=altura

    def calcularArea(self):
        print(f"Area de la rectangulo {self.base*self.altura}\n")

class Triangulo(Figura):
    def __init__(self,base,altura):
        self.base=base
        self.altura=altura

    def calcularArea(self):
        print(f"Area de la triangulo {self.base*self.altura/2}\n")

if __name__ == "__main__":
    circulo=Circulo(5)
    triangulo=Triangulo(3,4,)
    rectangulo=Rectangulo(3,5)
    figuras = [circulo,triangulo,rectangulo]

    for figura in figuras:
        figura.calcularArea()
